### PR TITLE
201: Implement login screen

### LINKED ISSUE #209: Functional Requirements: Intro Screen
When the app is opened for the first time, a small slideshow on the app's functions can be run through.

In a login and intro screen, users can “Log in now” or “Explore the app”. After going through up to 5 slides, users are equally taken to the login screen. In addition, links to the privacy policy and imprint are displayed on the login and intro screen.

The page also contains a “Support” button, which redirects users (in a browser tab) to the donation page, and a button for “Guest login with invitation code”.

Tasks:
- #102 

grafik
grafik
grafik

### PR DESCRIPTION
**Pull request by steffenkleinle**
### Short description

Implement the login screen according to the design including (so far empty) bottom sheet.

### Proposed changes

- Move auth files to `login`
- Add buttons and UI elements for login
- Add bottom sheet

### Side effects

- None.

### Testing

Make sure that you are not logged in yet and try the login button and the bottom action sheet.
Logging out is implemented in #324.

### Resolved issues

Fixes: #201
Partly fixes #209 

### Additional information

---

### CODE REVIEW

File: lib/features/login/widgets/welcome_view.dart [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:gruene_app/app/theme/theme.dart';
+import 'package:gruene_app/features/login/bloc/auth_bloc.dart';
+import 'package:gruene_app/i18n/translations.g.dart';
+
+class WelcomeView extends StatelessWidget {
```
Comment (steffenkleinle) [inline lib/features/login/widgets/welcome_view.dart:8]: Not sure about whether to extract this to an own widget and if yes, on which naming to use. Do you have any suggestions?

File: lib/features/login/bloc/auth_bloc.dart [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import 'package:flutter_bloc/flutter_bloc.dart';
 
-import 'package:gruene_app/features/auth/repository/auth_repository.dart';
+import 'package:gruene_app/features/login/repository/auth_repository.dart';
```
[the remark this answers is not in the document]
Comment (volkramweber) [reply lib/features/login/bloc/auth_bloc.dart:3]: @steffenkleinle :thought_balloon: I'm really not sure about this change. The AuthRepository also contains actions for logout and get the access token. It's not only used at login screen. So I think the feature name `auth` was the right one?

[the remark this answers is not in the document]
Comment (steffenkleinle) [reply lib/features/login/bloc/auth_bloc.dart:3]: Makes sense, I agree. Then I would maybe move it to `app/auth` though as it is not a feature in the way the other features are (for example it does not have a screen) :thinking: What do you think?

[the remark this answers is not in the document]
Comment (volkramweber) [reply lib/features/login/bloc/auth_bloc.dart:3]: :+1: sounds good!

**Comment by volkramweber** [comment]
:upside_down_face: The dot between data privacy and imprint looks a little smaller as shown in the figma...? Also the text "Erfahre hier, was Dich erwartet" is not visible on my phone until I open the bottom action sheet.

image
image

**Comment by steffenkleinle** [comment]
> 🙃 The dot between data privacy and imprint looks a little smaller as shown in the figma...?

Good catch! This was actually because I used a material circle icon but remove that from dependencies again. Changed it to a custom container, should be fixed.

> Also the text "Erfahre hier, was Dich erwartet" is not visible on my phone until I open the bottom action sheet.

Increased the initial bottom sheet height, but perhaps we have to find some other way in the future to assert it looks best on all devices.

**Review by volkramweber — APPROVED** [review]

**Pull request merged**